dev/NavigationView/NavigationView.cpp

```diff
@@ -740,22 +740,33 @@ void NavigationView::RaiseItemInvokedForNavigationViewItem(const winrt::Navigati
 
 void NavigationView::OnNavigationViewItemInvoked(const winrt::NavigationViewItem& nvi)
 {
-    auto selectedItem = SelectedItem();
-    RaiseItemInvokedForNavigationViewItem(nvi);
+    m_shouldRaiseItemInvokedAfterSelection = true;
 
-    // User changed selectionstate in the ItemInvoked callback
-    if (selectedItem != SelectedItem())
+    auto scopeGuard = gsl::finally([this]()
     {
-        return;
-    }
+        m_shouldRaiseItemInvokedAfterSelection = false;
+    });
 
+    auto selectedItem = SelectedItem();
     bool updateSelection = m_selectionModel && nvi.SelectsOnInvoked();
     if (updateSelection)
     {
         auto ip = GetIndexPathForContainer(nvi);
         UpdateSelectionModelSelection(ip);
     }
 
+    // Item was invoked but already selected, so raise event here.
+    if (selectedItem == nvi)
+    {
+        selectedItem = SelectedItem();
+        RaiseItemInvokedForNavigationViewItem(nvi);
+        // User changed selectionstate in the ItemInvoked callback
+        if (selectedItem != SelectedItem())
```

Comment:
This check should be moved to `ChangeSelection`.

The purpose of this check is to make sure we don't get into a bad selection state if user updates selecteditem in the iteminvoked callback.

Reply:
Hmm I see. I've updated `ChangeSelection`  to return if we notice that the selection changed during the ItemInvoked event processing.